### PR TITLE
Add wipe clustering command on reset...

### PR DESCRIPTION
**Pull request by matiasdelellis**
Generalize the ResetAll command to add more options. Now is just Reset, and you must to pass --all argument to reset everything as until now, or --clustering to just drop clusters..

The command is now face:reset and need an argument about what you want to reset.
--all, to delete everything, or --clustering to reset persons.

### CODE REVIEW

File: lib/Db/FaceMapper.php [hunk 1]
```diff
@@ -175,6 +198,26 @@ public function deleteUserFaces(string $userId) {
...
+	/**
+	 * Unset relation beetwen faces and persons from that user in order to reset clustering
+	 *
+	 * @param string $userId User to drop fo unset relation.
+	 */
+	public function unsetPersonsRelationForUser(string $userId) {
+		$sub = $this->db->getQueryBuilder();
+		$sub->select(new Literal('1'));
+		$sub->from('facerecog_images', 'i')
+			->where($sub->expr()->eq('i.id', '*PREFIX*' . $this->getTableName() .'.image'))
```
Comment (stalker314314) [inline lib/Db/FaceMapper.php]: should you filter here by model? As a user, I would be annoyed if I do `reset` and it wipes clustering of my non-current model:)

File: lib/Db/FaceMapper.php [hunk 2]
```diff
@@ -175,6 +198,26 @@ public function deleteUserFaces(string $userId) {
 			->execute();
 	}
 
+	/**
+	 * Unset relation beetwen faces and persons from that user in order to reset clustering
+	 *
+	 * @param string $userId User to drop fo unset relation.
+	 */
+	public function unsetPersonsRelationForUser(string $userId) {
```
Comment (stalker314314) [inline lib/Db/FaceMapper.php]: I might be stupid, but I don't see point of setting this to NULL when you are calling `deleteUserPersons` after you call this? Resetting clustering is done by deleting all entries in `persons` table for a given user, why you need this?

File: lib/Command/ResetCommand.php [hunk 3]
```diff
@@ -87,9 +101,27 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
+			$output->writeln('Reset successfully done');
+			return 0;
+		}
+		else if ($input->getOption('clustering')) {
+			$this->resetClusters($user);
+			$output->writeln('Reset clustering done');
+			return 0;
+		}
+		else {
+			$output->writeln('You must specify that you want to reset...');
```
[the remark this answers is not in the document]
Comment (matiasdelellis) [reply lib/Command/ResetCommand.php]: MM.. No .. It is not relevant for any user... and probably either not for the administrator who installs the application. Maybe it is only useful for our test.. (Or maybe my experiments. haha).

File: lib/Db/FaceMapper.php (hunk 1 above)
Comment (matiasdelellis) [reply lib/Db/FaceMapper.php]: Ohh.. You're right!

File: lib/Db/FaceMapper.php (hunk 2 above)
Comment (matiasdelellis) [reply lib/Db/FaceMapper.php]: Honestly, just to make it more real. The command option resets the clusters... How is the database before clustering? It has the column `person` as NULL. Therefore, I tried to replicate this.

File: lib/Db/FaceMapper.php (hunk 1 above)
Comment (matiasdelellis) [reply lib/Db/FaceMapper.php]: The same applies to delete persons, but I am seeing that we (Probably just me) ignore the consideration of the model in many PersonMapper queries. :disappointed: 

That said, there will be a future PR when implement other models. :disappointed:

**Pull request merged**